### ISSUE TITLE
proposal: implement the v0.5 transport

### ISSUE THREAD

**Issue by CAFxX** [DataDog/dd-trace-go#764]
datadog-agent [recently added support for a new trace ingestion endpoint](https://github.com/DataDog/datadog-agent/blob/master/CHANGELOG.rst#7220--6220) that is expected to be especially beneficial to lower memory usage and possibly cpu usage of the agent:

> APM: A new trace ingestion endpoint was introduced at /v0.5/traces which supports a more compact payload format, greatly improving resource usage. The spec for the new wire format can be viewed here. Tracers supporting this change will automatically use the new endpoint.

IIUC this is [not yet supported by dd-trace-go](https://github.com/DataDog/dd-trace-go/blob/977bbb89f54010b920264aeda051b5a67f46c26a/ddtrace/tracer/transport.go#L101). As these benefits can not be obtained until the clients start using the new transport, it would be ideal if support for the new transport was implemented soon - as our agents are frequently a source of pain regarding resource usage.

**Comment by gbbr** [comment]
This is true. We should also implement this new endpoint ([spec](https://github.com/DataDog/datadog-agent/blob/7.22.0/pkg/trace/api/version.go#L21)).

**Comment by johejo** [comment]
Implementation in other dd-trace
- https://github.com/DataDog/dd-trace-java/pull/1679
- https://github.com/DataDog/dd-trace-js/pull/1052

What is the priority of this issue within the dd-trace-go maintainer?
Hot or waiting for the community to contribute?

**Comment by knusbaum** [comment]
@johejo We can't give a timeline for this yet, but it is something we are very aware of.
If the community is able to contribute, that is always welcome. 

I will note the user interest in this and make sure the priority is set appropriately.

**Comment by gbbr** [comment]
Since this requires several changes to the code structure to allow creating different payloads (perhaps using the same transport structure), let's discuss the approach here before committing to writing code, if possible.

**Comment by knusbaum** [comment]
We'll look at this in future planning.

**Comment by darccio** [comment]
Closing this in favour of our current work implementing v1.0 transport (see PR #3932)
